scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb

```diff
@@ -48,12 +54,14 @@ namespace <%= namespace %>.SDK {
         public string ID = null;
         private string _VariantId;
         private long _Quantity;
+        private decimal _Price;
         private ObservableDictionary<string, string> _CustomAttributes;
         private OnCartLineItemChange OnChange;
-        
-        public CartLineItem(string variantId, OnCartLineItemChange onChange, long quantity = 1, IDictionary<string, string> customAttributes = null) {
+
+        public CartLineItem(string variantId, OnCartLineItemChange onChange, decimal price, long quantity = 1, IDictionary<string, string> customAttributes = null) {
```

Comment:
> This feels kind of shaky since we give power to the user to enter their own price, which is always a bad idea.

Definitely agree with you on this point.

So we've got three overloads of AddOrUpdate:

```public void AddOrUpdate(string variantId, long? quantity = null, Dictionary<string, string> customAttributes = null)

public void AddOrUpdate(ProductVariant variant, long? quantity = null, Dictionary<string, string> customAttributes = null)

public void AddOrUpdate(Product product, Dictionary<string, string> selectedOptions, long? quantity = null, Dictionary<string, string> customAttributes = null)
```

The later two are defined as basically calling through to the first version, where we actually use the `variantId`. I checked through our examples and, in all the cases I saw, we are recommending that developers use the second version. This makes the most sense too, imho, since when developers query products they get back a list of variants, so its reasonable they then takes those ProductVariant instances, and pass them back to Cart. 

With that in mind, I think we have two refactors here:

1. We make `public void AddOrUpdate(string variantId, long? quantity = null, Dictionary<string, string> customAttributes = null)` private (or remove it entirely). 
2. We change `CartLineItem` to accept a `ProductVariant`, instead of just an id.

@sleroux @dengjeffrey If you guys are good with that, I can do that as a separate, precursor PR to this.